File: python/raindrops/raindrops.py
def convert(number):
    answer = ''

    if number % 3 == 0:
        answer += 'Pling'
    if number % 5 == 0:
        answer += 'Plang'
    if number % 7 == 0:
        answer += 'Plong'

    if not answer:
        answer = str(number)

    return answer

File: erlang/raindrops/src/raindrops.erl
-module(raindrops).

-export([convert/1]).


convert(Number) ->
  Items = [{3, "Pling"}, {5, "Plang"}, {7, "Plong"}],

  Result = lists:foldl(
    fun (Tuple, Acc) ->
      make_sound(Tuple, Acc, Number) end,
    "",
    Items
  ),

  case Result of
    [] -> integer_to_list(Number);
    _ -> Result
  end
.

make_sound({Item, Sound}, Acc, Number)
  when Number rem Item =:= 0
  ->
    Acc ++ Sound
;
make_sound(_, Acc, _) ->
  Acc
.
